### PR TITLE
Hardening the session store saving code

### PR DESCRIPTION
**Pull request by bsclifton**
Should help with https://github.com/brave/browser-laptop/issues/11576

Auditors: @darkdh

Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: test/unit/app/sessionStoreTest.js [hunk 1]
```diff
@@ -159,6 +159,50 @@ describe('sessionStore unit tests', function () {
...
+    it('throws an error when input is mutable', function () {
+      assert.throws(() => {
+        sessionStore.cleanPerWindowData({windowStuff: 'demo'})
+      },
+      /undefined == true/,
+      'did not throw with expected message')
+    })
+    describe('when closed frames exist', function () {
+      it('works when closeFrames is empty', function () {
+        sessionStore.cleanPerWindowData(
```
Comment (darkdh) [inline test/unit/app/sessionStoreTest.js]: no result comparison?

File: test/unit/app/sessionStoreTest.js [hunk 2]
```diff
@@ -159,6 +159,50 @@ describe('sessionStore unit tests', function () {
...
+          Immutable.fromJS({
+            closedFrames: []
+          })
+        )
+      })
+      it('works when closedFrames array contains null', function () {
+        sessionStore.cleanPerWindowData(
```
Comment (darkdh) [inline test/unit/app/sessionStoreTest.js]: ditto

File: test/unit/app/sessionStoreTest.js [hunk 3]
```diff
@@ -159,6 +159,50 @@ describe('sessionStore unit tests', function () {
...
+              {key: 1, location: 'http://example.com'},
+              null,
+              {key: 2, location: 'https://brave.com'}
+            ]
+          })
+        )
+      })
+    })
+    it('works when closedFrames array contains null', function () {
+      sessionStore.cleanPerWindowData(
```
Comment (darkdh) [inline test/unit/app/sessionStoreTest.js]: ditto

File: test/unit/app/sessionStoreTest.js (hunk 1 above)
Comment (bsclifton) [reply test/unit/app/sessionStoreTest.js]: good call... let me clean this up a bit more...

Comment (bsclifton) [reply test/unit/app/sessionStoreTest.js]: (I wasn't checking value since the test fails when an error happens)

**Comment by bsclifton** [comment]
@darkdh updated! I also added additional tests 😄 👍

**Comment by bsclifton** [comment]
master https://github.com/brave/browser-laptop/commit/b8ac897bbba35cc333473468727c864745b3be71
0.21.x https://github.com/brave/browser-laptop/commit/11ef2f9d497f8f08d71f3263df9906436341e5c8
0.20.x https://github.com/brave/browser-laptop/commit/b4c53e9dea758673a5d10f1ddfd1c62784ea243c
0.19.x https://github.com/brave/browser-laptop/commit/4f309a0c0df2c716aacf94f3a555c2093612fb3b

**Pull request merged**